File: frontend/src/api/notifications.ts
import api from '@/api/client'

export interface NotificationItem {
  id: number
  title: string
  message?: string | null
  href?: string | null
  timestamp: string
  read: boolean
}

export interface NotificationListResponse {
  unread_count: number
  items: NotificationItem[]
}

export const listNotifications = async (limit = 20) => {
  const { data } = await api.get<NotificationListResponse>('/notifications', { params: { limit } })
  return data
}

export const markNotificationsRead = async () => {
  const { data } = await api.post('/notifications/mark-read')
  return data
}

File: backend/app/schemas/notifications.py
from datetime import datetime

from pydantic import BaseModel


class NotificationItem(BaseModel):
    id: int
    title: str
    message: str | None = None
    href: str | None = None
    timestamp: datetime
    read: bool


class NotificationListResponse(BaseModel):
    unread_count: int
    items: list[NotificationItem]
